adapter/test/vfd/CMakeLists.txt

```diff
@@ -4,36 +4,64 @@ set(hermes_vfd_tests
   hermes_vfd_test
 )
 
-foreach(vfd_test ${hermes_vfd_tests})
-  add_executable(${vfd_test} ${CMAKE_CURRENT_SOURCE_DIR}/${vfd_test}.cc)
-  add_dependencies(${vfd_test} hermes)
-  target_include_directories(${vfd_test} PRIVATE ${HERMES_VFD_DIR})
-  target_include_directories(${vfd_test}
-    SYSTEM PUBLIC ${HDF5_HERMES_VFD_EXT_INCLUDE_DEPENDENCIES}
-  )
-  target_link_libraries(${vfd_test}
-    MPI::MPI_CXX
-    hermes
-    ${HDF5_HERMES_VFD_EXPORTED_LIBS}
-    ${HDF5_HERMES_VFD_EXT_LIB_DEPENDENCIES}
-    ${HDF5_HERMES_VFD_EXT_PKG_DEPENDENCIES}
-  )
-  add_test(NAME "Test${vfd_test}" COMMAND ${vfd_test})
-  set_property(TEST "Test${vfd_test}"
-    PROPERTY ENVIRONMENT HERMES_CONF=${CMAKE_CURRENT_SOURCE_DIR}/hermes.conf)
-  set_property(TEST "Test${vfd_test}" APPEND
+add_executable(hermes_vfd_test ${CMAKE_CURRENT_SOURCE_DIR}/hermes_vfd_test.cc)
+target_include_directories(hermes_vfd_test PRIVATE ${HERMES_VFD_DIR})
+target_include_directories(hermes_vfd_test PRIVATE ${HERMES_ADAPTER_TEST_DIR})
+target_include_directories(hermes_vfd_test
+  SYSTEM PRIVATE ${HDF5_HERMES_VFD_EXT_INCLUDE_DEPENDENCIES}
+)
+
+target_link_libraries(hermes_vfd_test
+  hermes
+  MPI::MPI_CXX
+  stdc++fs
+  ${HDF5_HERMES_VFD_EXT_LIB_DEPENDENCIES}
+  ${HDF5_HERMES_VFD_EXT_PKG_DEPENDENCIES}
+)
+
+if(HERMES_USE_ADDRESS_SANITIZER)
+  execute_process(COMMAND ${CMAKE_C_COMPILER} -print-file-name=libasan.so
+    OUTPUT_VARIABLE LIBASAN_PATH
+    RESULT_VARIABLE ASAN_PRINT_FILE_NAME_RESULT
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  if(ASAN_PRINT_FILE_NAME_RESULT EQUAL 0)
+    message(STATUS "Found libasan.so at ${LIBASAN_PATH}")
+  else()
+    message(FATAL_ERROR
+      "Couldn't find the path to libasan.so which is required by the Hermes HDF5 VFD tests. \
+       Recompile with HERMES_ENABLE_ADDRESS_SANITIZER=OFF")
+  endif()
+endif()
+
+function(set_vfd_test_properties test_name)
+  set_property(TEST ${test_name}
+    PROPERTY ENVIRONMENT HERMES_CONF=${HERMES_ADAPTER_TEST_DIR}/data/hermes.conf)
+  set_property(TEST ${test_name} APPEND
+    PROPERTY ENVIRONMENT HDF5_PLUGIN_PATH=${HERMES_VFD_LIBRARY_DIR})
+  set_property(TEST ${test_name} APPEND
+    PROPERTY ENVIRONMENT HDF5_DRIVER=hermes)
+  set_property(TEST ${test_name} APPEND
+    PROPERTY ENVIRONMENT LD_PRELOAD=${LIBASAN_PATH}:$<TARGET_FILE:hdf5_hermes_vfd>)
+  set_property(TEST ${test_name} APPEND
     PROPERTY ENVIRONMENT LSAN_OPTIONS=suppressions=${CMAKE_SOURCE_DIR}/test/data/asan.supp)
-endforeach()
+endfunction()
+
+add_test(NAME "TestVfd" COMMAND hermes_vfd_test --reporter compact -d yes)
```

Comment:
Is there a reason for using Vfd instead of VFD?

Reply:
1. It's easier to type.
2. I tend to use that convention because when you capitalize an acronym in camel case the acronym runs into the next word and is harder to read in my opinion. For example, in `VFDTest` the D and T run together whereas `VfdTest` has a clear separation between the two words.